examples/webgl_lightshafts.html

```diff
@@ -0,0 +1,249 @@
+
+			}
+
+			var container, controls, clock;
+			var camera, scene, renderer, uniforms;
+
+			init();
+			animate();
+
+			function init() {
+
+				container = document.createElement( 'div' );
+				document.body.appendChild( container );
+
+				camera = new THREE.PerspectiveCamera( 45, window.innerWidth / window.innerHeight, 0.1, 100 );
+				camera.position.set( 3.2, 3, 3.7 );
+
+				controls = new THREE.OrbitControls( camera );
+				controls.target.set( 0.5, 1.5, 0 );
+				controls.minDistance = 2;
+				controls.maxDistance = 20;
+				controls.update();
+
+				clock = new THREE.Clock();
+
+				scene = new THREE.Scene();
+				scene.background = new THREE.Color( 0xcf8b74 );
+
+				var	light = new THREE.HemisphereLight( 0xffffff, 0x444444 );
+				light.position.set( 0, 20, 0 );
+				scene.add( light );
+
+				light = new THREE.DirectionalLight( 0xffffff );
+				light.position.set( 0, 20, 10 );
+				scene.add( light );
+
+				var loader = new THREE.GLTFLoader().setPath( 'models/gltf/' );
+				loader.load( 'tree.glb', function ( gltf ) {
+
+					gltf.scene.traverse( function ( child ) {
+
+						if ( child.isMesh )	{
+
+							child.material.transparent = false;
+							child.material.alphaTest = 0.5;
+
+						}
+
+					} );
+
+					scene.add( gltf.scene );
+
+				} );
+
+				// rays
+
+				var textureLoader = new THREE.TextureLoader();
+				var texture = textureLoader.load( 'textures/lightShaft.png' );
+
+				uniforms = {
+					// controls how fast the ray attenuates when the camera comes closer
+					attenuation: {
+						value: 10
+					},
+					// controls the speed of the animation
+					speed: {
+						value: 2
+					},
+					// the color of the ray
+					color: {
+						value: new THREE.Color( 0xdadc9f )
+					},
+					// the visual representation of the ray highly depends on the used texture
+					texture: {
+						value: texture
+					},
+					// global time value for animation
+					time: {
+						value: 0
+					},
+					// individual time offset so rays are animated differently if necessary
+					timeOffset: {
+						value: 0
+					}
+				};
+
+				var lightShaftMaterial = new THREE.ShaderMaterial( {
+					uniforms: uniforms,
+					vertexShader: document.getElementById( 'vertexShader' ).textContent,
+					fragmentShader: document.getElementById( 'fragmentShader' ).textContent,
+					blending: THREE.AdditiveBlending,
+					depthWrite: false,
+					transparent: true,
+					side: THREE.DoubleSide
+				} );
+
+				var lightShaftGeometry = new THREE.PlaneBufferGeometry( 0.5, 5 );
+
+				for ( var i = 0; i < 5; i ++ ) {
+
+					var lightShaft = new THREE.Mesh( lightShaftGeometry, lightShaftMaterial );
+					lightShaft.position.x = - 1 + 1.5 * Math.sign( ( i % 2 ) );
+					lightShaft.position.y = 2;
+					lightShaft.position.z = - 1.5 + ( i * 0.5 );
+					lightShaft.rotation.y = Math.PI * 0.2;
+					lightShaft.rotation.z = Math.PI * - ( 0.15 + 0.1 * Math.random() );
+					scene.add( lightShaft );
```

Comment:
Minor nit: maybe add this inside the onLoad function so that everything appears at once? 

On my mobile I'm staring at the light shafts for a couple of seconds before the model loads, it looks a bit off. 

Lovely demo BTW 😁

Reply:
That's a good point 👍